### PR TITLE
Update System before testing

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

**Comment by coolo** [comment]
This is trying to fix https://progress.opensuse.org/issues/13058 - I only looked at the xfce case so far though. Needs more testing

File: products/opensuse/main.pm [hunk 1]
```diff
@@ -748,6 +737,25 @@ sub install_online_updates {
...
+        loadtest "update/prepare_system_for_update_tests.pm";
+        if (check_var("DESKTOP", "kde")) {
+            loadtest "update/updates_packagekit_kde.pm";
+        }
+        else {
+            loadtest "update/updates_packagekit_gpk.pm";
+        }
+    }
+    else {
+        if (!guiupdates_is_applicable()) {
```
Comment (kravciak) [inline products/opensuse/main.pm]: Isn`t this condition already checked by "else"?

**Comment by kravciak** [comment]
sle/main.pm also needs to be updated as test paths changed to tests/update/

File: products/opensuse/main.pm (hunk 1 above)
Comment (coolo) [inline products/opensuse/main.pm]: yes, sure. I was just craming the pieces together

**Comment by coolo** [comment]
But https://tortuga.suse.de/tests/394 shows it works for xfce - now on to fix the rest

**Comment by okurz** [comment]
Is this a real PR or just some scratch rework? The commits don't really tell the benefit or the purpose just stating what is done. Also make sure to use your name and email for commits and not openQA webUI.

**Comment by coolo** [comment]
this branch is updated from my staging server to fix opensuse maintenance. So it's not a real PR - especially not a done one.

**Comment by coolo** [comment]
https://tortuga.suse.de/tests/405 and 406 for textmode and gnome

**Comment by okurz** [comment]
LGTM

**Comment by DimStar77** [comment]
https://openqa.opensuse.org/tests/239783#step/updates_packagekit_kde/11

As this now runs before consoletest_setup, the workarounds implemented there to fixup network post updates are missing, causing the test not to have network connect

**Comment by coolo** [comment]
hmm, you say there are bugs in TW that require workarounds in our tests?

**Comment by DimStar77** [comment]
No - I say they are bugs in our released products.

in consoletest_setup, we had this piece of code:

>     # openSUSE 13.2's (and earlier) systemd has broken rules for virtio-net, not applying predictable names (despite being configured)
>     # A maintenance update breaking networking names sounds worse than just accepting that 13.2 -> TW breaks
>     # At this point, the system has been upadted, but our network interface changes name (thus we lost network connection)
>     my @old_hdds = qw/openSUSE-12.1 openSUSE-12.2 openSUSE-12.3 openSUSE-13.1-gnome openSUSE-13.2/;
>     if (grep { check_var('HDDVERSION', $_) } @old_hdds) {    # copy eth0 network config to ens4
>         script_run("cp /etc/sysconfig/network/ifcfg-eth0 /etc/sysconfig/network/ifcfg-ens4");
>         script_run("/sbin/ifup ens4");
>     }
>

**Comment by coolo** [comment]
ok, but then these workarounds should move to prepare_system_for_update or even better to happen before first_boot IMO.

**Comment by nilxam** [comment]
hmm... can't we just do not go through load_system_update_tests() if UPGRADE=1 or ZDUP=1?

**Comment by coolo** [comment]
fine with me to just disable it in these cases

**Comment by nilxam** [comment]
JFYI, I knew it's not about this change :)
the interesting thing we seen now is https://openqa.opensuse.org/tests/240488#step/updates_packagekit_kde/13 failure

it's a 42.1-to-staging-update test, updater icon does not in the tray as expected, but after pkcon refresh, PK found something need to update.. we have to output PK's update list then...

**Comment by kravciak** [comment]
I think update icon shows after refreshing repositories, which was done in console tests before this PR

**Comment by DimStar77** [comment]
right - and the 42.1 update case in staging has the side effect that:

* 42.1 image is updated using the staging dvd - as usual, the DVD is an incomplete subset of packages
* after reboot, PK did not yet refresh the repo list, so there are no updates 'available'
* we call pkcon refresh, followed by pkcon get-updates and expect there not to  be updates available.

I think the prepare-system-for-updates tasks should take care of ensuring the repos are refreshed already at that point.

@coolo ?

**Comment by coolo** [comment]
sounds reasonable.  Also this check that @kravciak added to the kde test is missing for the other scenarios. So possibly we want to move that to a 3rd test run after the actual update

**Comment by kravciak** [comment]
working on it in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/1692

**Pull request merged**